pylint/checkers/classes/class_checker.py

```diff
@@ -1950,8 +1950,13 @@ def _check_first_arg_for_type(
             # Check if there is a decorator which is not named `staticmethod` but is assigned to one.
             return
         # class / regular method with no args
-        elif not node.args.args and not node.args.posonlyargs:
-            self.add_message("no-method-argument", node=node)
+        elif (
+            not node.args.args
+            and not node.args.posonlyargs
+            and not node.args.vararg
+            and not node.args.kwarg
```

Comment:
could this be refactored to remove so many `not` to
```(
not (node.args.args and node.args.pos...... and ... and )
```

maybe not? worth checking

Reply:
Yeah, that makes sense, thanks.